google-cloud-error_reporting/lib/google/cloud/error_reporting/middleware.rb

```diff
@@ -73,7 +74,7 @@ def initialize app, error_reporting: nil, project_id: nil,
                         ENV["GOOGLE_CLOUD_PROJECT"] ||
                         Google::Cloud::Core::Environment.project_id
 
-          raise ArgumentError, "project_id is required" if @project_id.nil?
+          fail ArgumentError, "project_id is required" if @project_id.nil?
```

Comment:
I had the same question before and after some reading, I found this seems to be a controversial topic within the Ruby community on when to use fail vs raise.

My favorite explanation is this [stack overflow answer](http://stackoverflow.com/questions/31937632/fail-vs-raise-in-ruby-should-we-really-believe-the-style-guide), where raise is only to be used within a catch block. Every where else uses fail to indicate an exception.